packages/opencensus-core/src/trace/model/tracer.ts

```diff
@@ -46,7 +46,14 @@ export class CoreTracer implements types.Tracer {
   sampler: samplerTypes.Sampler;
   /** A configuration for starting the tracer */
   logger: loggerTypes.Logger = logger.logger();
-
+  /** maxAnnotationEvents is max number of annotation events per span  */
+  numberOfAnnontationEventsPerSpan?: number;
```

Comment:
Rather than storing the individual values here, could it just store something like `traceParameters: TraceParameters = DEFAULT_TRACE_PARAMETERS`?

Then if we add one more parameter later on, we don't need to add it two places.

Reply:
I have a question, do we assume that user configures the trace with parameters having values more than their limit?

Reply:
Ah, I missed the logic of the builder pattern below that capped those values at their limits. If other languages have that cap, we should preserve it for Node I'd think.

In that case, I would recommend making these be top-level attributes (like you have) but be made `readonly` to prevent changes to them once they are initialized.